lib/arclight/traject/ead2_config.rb

```diff
@@ -126,6 +126,14 @@
   accumulator.replace([accumulator.any?])
 end
 
+to_field 'names_coll_ssim', extract_xpath('/xmlns:ead/xmlns:archdesc[@level="collection"]/xmlns:controlaccess', to_text: false) do |_record, accumulator|
```

Comment:
Couple notes on the xpath here:

- uses root-relative expression to get to the `archdesc` rather than `//` as `archdesc` [can only be a child of `ead`](https://www.loc.gov/ead/tglib/elements/archdesc.html). I would imagine we may see performance gains by not using `//` if the element has a predictable path.
- uses `level="collection"` attribute for specifically targeting collection properties. `@level` is required, but [could be something other than `collection`](https://www.loc.gov/ead/tglib/elements/archdesc.html).